### ISSUE TITLE
Investigate `create-env` as a solution for single-instance Concourse deployments

### ISSUE THREAD

**Issue by vito** [concourse/concourse#1152]
How close can we get this to `vagrant up`? It would make our lives easier by removing a distribution format, and could act as a gateway drug into BOSH. It'd also let you reconfigure and redeploy it, something `vagrant up` doesn't let you do.

**Comment by vito** [comment]
#1006 would greatly help this

**Comment by pivotal-topher-bullock** [comment]
After some investigation one gotcha I ran into was around BOSH links being used in control scripts. Since there's no director managing deployment(s), there's nothing there to do the link-ing. Good news is this can mostly be avoided by providing explicit values for things that are usually resolved as links.

Concourse releases need to be compiled (#1006); and we'll also need a compiled release of Garden-runC to be able to install everything on a single vm.

**Comment by pivotal-topher-bullock** [comment]
Investigated more with @xtreme-gavin-enns and collected our findings here : https://github.com/concourse/paper-plane 

**TLDR:** compiled releases ( via `bosh export-release`) aren't a hard requirement, but speed things up a lot. Pre-compiling the packages (#1006) would make the process a lot faster.. compiling golang, postgresql, tar, etc can take a lot of time.

**Comment by xtreme-gavin-enns** [comment]
yeah, it was like 3 min with pre-compiled releases (not including the cpi, iirc) versus about 10 without

**Comment by vito** [comment]
This is about as investigated as we need it to be; opened #1380 for the next steps.

**Comment by xtreme-gavin-enns** [comment]
Beware: It is not clear if it is possible to use create-env to spin up a single-vm deployment other than a bosh director on Azure or AWS. This is because those iaases do not support storing some metadata that bosh requires and therefore a bosh director registry job is required.
